hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroIndexedRecord.java

```diff
@@ -185,9 +188,17 @@ public HoodieRecord joinWith(HoodieRecord other, Schema targetSchema) {
   }
 
   @Override
-  public HoodieRecord prependMetaFields(Schema recordSchema, Schema targetSchema, MetadataValues metadataValues, Properties props) {
+  public HoodieRecord prependMetaFields(Schema recordSchema, Schema targetSchema, MetadataValues metadataValues, Properties props,
+                                        boolean hasOperationMetaField) {
     decodeRecord(recordSchema);
-    GenericRecord newAvroRecord = HoodieAvroUtils.rewriteRecordWithNewSchema(data, targetSchema);
+    List<String> metaFields = hasOperationMetaField ? HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION_LIST : HoodieRecord.HOODIE_META_COLUMNS;
```

Comment:
The `targetSchema` already includes the `_hoodie_operation` field if it is enabled, you can just use the `HoodieAvroUtils.isMetadataField` to filter out the metadata fields. Or just check whether the 6th field is `_hoodie_operation` to see it if it is enabled.